vaadin-spring/src/main/java/com/vaadin/flow/spring/SpringBootAutoConfiguration.java

```diff
@@ -82,11 +82,16 @@ public ServletContextInitializer contextInitializer() {
                 new SpringServlet(context, rootMapping), mapping);
         registration.setInitParameters(initParameters);
         registration
-                .setAsyncSupported(configurationProperties.isAsyncSupported());
+                .setAsyncSupported(
+                        configurationProperties.getAsyncSupported() == null ?
+                                true : // default value
+                                configurationProperties.getAsyncSupported());
         registration.setName(
                 ClassUtils.getShortNameAsProperty(SpringServlet.class));
         registration.setLoadOnStartup(
-                configurationProperties.isLoadOnStartup() ? 1 : -1);
+                configurationProperties.getLoadOnStartup() == null ||
```

Comment:
Same question here: should the return type be `boolean` and not `Boolean`
And then the getter name can be preserved as `isXXX`

Reply:
more or less the same as before. I can implement a more complex code generation logic if you think it's worthed

Reply:
The point is : API is changed just because of code generation simplicity.

See above.

Reply:
same as above. Though I see this code is missing, now.